platform/podjs/src/browserPod.ts

```diff
@@ -92,11 +121,15 @@ class StorePolyIn implements PolyIn {
     }
 
     async add(quad: RDF.Quad): Promise<void> {
-        (await this.store).add(quad);
+        const store = await this.store;
+        store.add(quad);
+        await this.sync(store);
```

Comment:
Is it required to sync on any write operation? Wouldn't it be too costly if the graph is big enough?

What about syncing the store only when PodJS is closed(aka when user session ends)?

Reply:
> Is it required to sync on any write operation? Wouldn't it be too costly if the graph is big enough?

Correct, this will be slow for large data sets. But it's the best we can do with Oxigraph's API. Ideally, we would save each quad individually in IndexedDB and only update the quads that changed (like we did before) but Oxigraph doesn't provide hooks to respond to individual quads changing.

> What about syncing the store only when PodJS is closed(aka when user session ends)?

I don't think that is possible with a web-based pod implementation. The `unload` and `beforeunload` events are not fired in a all scenarios. Also since writing the data to IndexedDB is an asynchronous operation it wouldn't be guaranteed to complete before the document is closed. We could possibly delegate IO to a Service Worker, but Service Workers aren't available in documents served from a `file:` URL.

I will be going ahead merging this PR into the `oxigraph-pod-js` branch which is still WIP.  We can continue this discussion in https://jira.polypoly.eu/browse/PROD4POD-1811.